### ISSUE TITLE
Lets work on the "Support more raspberry pi than rpi3" goal

### ISSUE THREAD

**Issue by joelbyler** [jjcarstens/replex#2]
I'd love for this to work on rpizero[w], [rpitx](https://github.com/F5OEO/rpitx) claims to support this form factor.  Any thoughts on how this would work and what might need changed?

**Comment by joelbyler** [comment]
FWIW, when trying with the current build against `MIX_TARGET=rpi0` I get the following error

```
scrub-otp-release.sh: ERROR: Unexpected executable format for '/.../my_app/_build/_nerves-tmp/rootfs-additions/srv/erlang/lib/replex-0.1.0/priv/sendiq_rpi3'

Got:
 readelf:ARM;0x5000400, Version5 EABI, hard-float ABI

Expecting:
 readelf:ARM;0x5000200, Version5 EABI, soft-float ABI

This file was compiled for the host or a different target and probably
will not work.
```

**Comment by jjcarstens** [comment]
Thanks @joelbyler!

For some reason, rpi0 “compiles”, but the file is significantly smaller and non-executable so something is happening during compiling that is not being exposed. I’m looking at that a bit today.

In the meantime, we should probably compile on raspbian to see if the result is any different. And if we can also get the rpitx source compiling on rpi0, we can try putting the resulting `sendiq` binary in a nerves fw and see if it works (if I remember correctly, that had other errors..may be a clue to why crosscompiling isn’t working)

**Comment by joelbyler** [comment]
@jjcarstens thanks for checking into this, I’m headed to work today but can try this out on raspian this evening with a rpi0 and see if I can get it to build.  If that works, could I just fork your lib and try and swap out the binary?

**Comment by jjcarstens** [comment]
Actually, this might be simpler. I made a few changes to the make file and seem to be getting a valid binary now. Will PR soon but won't be able to test until later tonight

**Comment by joelbyler** [comment]
FYI, having some trouble with this still...  perhaps its a dependency that I'm missing?

```
$ MIX_TARGET=rpi0 mix firmware

Nerves environment
  MIX_TARGET:   rpi0
  MIX_ENV:      dev

==> replex
mkdir -p /Users/myuser/Workspaces/play/nerves_proj/_build/rpi0_dev/lib/replex/ebin/../priv
mkdir -p /Users/myuser/Workspaces/play/nerves_proj/_build/rpi0_dev/lib/replex/ebin/../obj
KCONFIG_NOTIMESTAMP=1 /usr/local/bin/gmake CROSS_COMPILE="/Users/myuser/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.2.0/bin/armv6-rpi-linux-gnueabi-" SED=/usr/local/bin/gsed -C /Users/myuser/Workspaces/play/nerves_proj/deps/replex//src/librpitx-5475c41ccf202b544cac6cf0c670ae40210a9f4b/src
gmake[1]: Entering directory '/Users/myuser/Workspaces/play/nerves_proj/deps/replex/src/librpitx-5475c41ccf202b544cac6cf0c670ae40210a9f4b/src'
#cc -Wall -O3 -Wno-unused-variable -c -o mailbox.o mailbox.c
cc -Wall -O3 -Wno-unused-variable -c -o raspberry_pi_revision.o raspberry_pi_revision.c
c++ -std=c++11 -Wall -O3 -Wno-unused-variable -c dsp.cpp iqdmasync.cpp ngfmdmasync.cpp fmdmasync.cpp  dma.cpp gpio.cpp serialdmasync.cpp phasedmasync.cpp amdmasync.h amdmasync.cpp fskburst.cpp ookburst.cpp atv.cpp util.cpp mailbox.c
clang: warning: treating 'c-header' input as 'c++-header' when in C++ mode, this behavior is deprecated [-Wdeprecated]
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
In file included from dma.cpp:25:
./mailbox.h:30:10: fatal error: 'linux/ioctl.h' file not found
#include <linux/ioctl.h>
         ^~~~~~~~~~~~~~~
1 error generated.
In file included from gpio.cpp:19:
./mailbox.h:30:10: fatal error: 'linux/ioctl.h' file not found
#include <linux/ioctl.h>
         ^~~~~~~~~~~~~~~
1 error generated.
atv.cpp:161:41: error: cast from pointer to smaller type 'unsigned int' loses information
    dbg_printf(1, "Last cbp :  %d \n", ((unsigned int)(cbp) - (unsigned int)(cbarray)) / sizeof(dma_cb_t));
                                        ^~~~~~~~~~~~~~~~~~~
atv.cpp:161:63: error: cast from pointer to smaller type 'unsigned int' loses information
    dbg_printf(1, "Last cbp :  %d \n", ((unsigned int)(cbp) - (unsigned int)(cbarray)) / sizeof(dma_cb_t));
                                                              ^~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
mailbox.c:40:10: fatal error: 'sys/sysmacros.h' file not found
#include <sys/sysmacros.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
gmake[1]: *** [Makefile:10: librpitx] Error 1
gmake[1]: Leaving directory '/Users/myuser/Workspaces/play/nerves_proj/deps/replex/src/librpitx-5475c41ccf202b544cac6cf0c670ae40210a9f4b/src'
make: *** [/Users/myuser/Workspaces/play/nerves_proj/_build/rpi0_dev/lib/replex/ebin/../obj/lib/librpitx.a] Error 2
could not compile dependency :replex, "mix compile" failed. You can recompile this dependency with "mix deps.compile replex", update it with "mix deps.update replex" or clean it with "mix deps.clean replex"
==> firmware
** (Mix) Could not compile with "make" (exit status: 2).
You need to have gcc and make installed. Try running the
commands "gcc --version" and / or "make --version". If these programs
are not installed, you will be prompted to install them.

$
```

**Comment by jjcarstens** [comment]
Just documenting that the last error here was solved with release v0.2.1
